user-interface/src/data-verification/ConsolidationOrderModal.tsx

```diff
@@ -33,72 +34,109 @@ export interface ConsolidationOrderModalProps {
+  return caseAssignments.map((assignment) => assignment.name);
+}
+
+export function addOxfordCommas(attorneys: string[]) {
+  if (attorneys.length === 0) {
+    return '(unassigned)';
+  } else if (attorneys.length < 3) {
+    return attorneys.join(' and ');
+  }
+  return attorneys.slice(0, -1).join(', ') + ', and ' + attorneys[attorneys.length - 1];
+}
+
 function ConsolidationOrderModalComponent(
   props: ConsolidationOrderModalProps,
   ConfirmationModalRef: React.Ref<ConfirmationModalImperative>,
 ) {
   const { id, onConfirm, onCancel }: ConsolidationOrderModalProps = props;
 
-  const modalRef = useRef<ModalRefType>(null);
-  const reasonRef = useRef<HTMLTextAreaElement>(null);
-  const [reason] = useState<string>('');
+  const [cases, setCases] = useState<ConsolidationOrderCase[]>([]);
+  const [childCasesDivHeight, setChildCasesDivHeight] = useState<string>('');
+  const [consolidationType, setConsolidationType] = useState<ConsolidationType | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [leadCaseAttorneys, setLeadCaseAttorneys] = useState<string[]>([]);
+  const [leadCaseDivisionCode, setLeadCaseDivisionCode] = useState<string>('');
+  const [leadCaseNumber, setLeadCaseNumber] = useState<string>('');
+  const [leadCaseNumberError, setLeadCaseNumberError] = useState<string>('');
+  const [leadCaseSummary, setLeadCaseSummary] = useState<CaseSummary | null>(null);
   const [options, setOptions] = useState<ShowOptions>({
     status: 'pending',
     heading: '',
-    attorneys: [],
   });
-  const [consolidationType, setConsolidationType] = useState<ConsolidationType | null>(null);
-  const [caseIds, setCaseIds] = useState<string[]>([]);
-  const [leadCaseDivisionCode, setLeadCaseDivisionCode] = useState<string>('');
-  const [leadCaseNumber, setLeadCaseNumber] = useState<string>('');
-  const leadCaseNumberRef = useRef<InputRef>(null);
-  const leadCaseDivisionRef = useRef<InputRef>(null);
+  const [reason] = useState<string>('');
+  const [step, setStep] = useState<ConfirmationSteps>('pick-lead-case');
+
   const administrativeConsolidationRef = useRef<RadioRef>(null);
+  const leadCaseDivisionRef = useRef<InputRef>(null);
+  const leadCaseNumberRef = useRef<InputRef>(null);
+  const modalRef = useRef<ModalRefType>(null);
+  const reasonRef = useRef<HTMLTextAreaElement>(null);
   const substantiveConsolidationRef = useRef<RadioRef>(null);
+
   const featureFlags = useFeatureFlags();
+  const windowSize = useWindowSize();
 
-  function clearReason() {
-    if (reasonRef.current) reasonRef.current.value = '';
-  }
+  const confirmStep2 = async () => {
+    onConfirm({
+      status: options.status,
+      rejectionReason: reasonRef.current?.value,
+      leadCaseSummary: leadCaseSummary!,
+      // onConfirm should never be called unless the button is enabled.
+      // The button should never be enabled unless a consolidationType is selected
+      consolidationType: consolidationType!,
+    });
```

Comment:
```suggestion
    if (!leadCaseSummary || !consolidationType) return; // or should we throw?
    onConfirm({
      status: options.status,
      rejectionReason: reasonRef.current?.value,
      leadCaseSummary: leadCaseSummary,
      consolidationType: consolidationType,
    });
```
I'd prefer to explicitly handle the scenario described in the comments.

Reply:
We should not throw. We have the lead case and the consolidation type as preconditions from the first step. This is handling the second "verify" step. We can put an explicit guard there and even throw just in case we break the logic in step 1 in the future.

Reply:
Loose the comments.